### ISSUE TITLE
azurerm_cognitive_account neither fails nor applies subnet as virtual network rule.

### ISSUE THREAD

**Issue by asishr** [hashicorp/terraform-provider-azurerm#13108]
### Terraform (and AzureRM Provider) Version

terraform -v = 1.0.5

provider registry.terraform.io/hashicorp/azurerm = v2.73.0

### Affected Resource(s)

* `azurerm_cognitive_account`

### Terraform Configuration Files

```hcl
resource "azurerm_cognitive_account" "example" {
  name                = "example-account"
  location            = "Canada Central"
  resource_group_name = "testResourceGroup"
  kind                = "Face"
  custom_subdomain_name = "testunit"
  sku_name = "S0"
  network_acls {
    default_action   = "Deny"
    ip_rules         = ["1.2.3.4"]
    virtual_network_rules = [{
      subnet_id = "/subscriptions/0000/resourceGroups/Network-RG/providers/Microsoft.Network/virtualNetworks/CirtualNetwork/subnets/subnettest" 
      ignore_missing_vnet_service_endpoint = false
    },{
      subnet_id = "/subscriptions/0000/resourceGroups/Network-RG/providers/Microsoft.Network/virtualNetworks/CirtualNetwork/subnets/subnettest1"
      ignore_missing_vnet_service_endpoint = false
    }]
  }
  tags = {
    Acceptance = "Test"
  }
}
```
### Expected Behaviour

Provided subnet in Subnet_id to be applied as Virtual Network Rule for cognitive account.

### Actual Behaviour

Provided subnet is not being applied nor failing in the terraform apply for cognitive account.
Apply is successful but Virtual Network rule is still empty in Azure.

### Steps to Reproduce

Did an apply from the above provided example

1. `terraform apply`

**Comment by brnwn4** [comment]
Im having this issue as well.

**Comment by samershr** [comment]
Same issue here. In Terraform tfstate there are different attributes than in Azure. Somehow Terraform ignores the `virtual_network_rules` while the second attribute `virtual_network_subnet_ids` does not exist in Azure.

tfstate:
```
"network_acls": [
  {
    "default_action": "Deny",
    ip_rules": [
       "*****"
    ],
    "virtual_network_rules": null,
     "virtual_network_subnet_ids": [„*****“]
  }
],
```

while in Azure:
```
"networkAcls": {
    "defaultAction": "Deny",
    "virtualNetworkRules": [
        {
            "id": "*****",
            "ignoreMissingVnetServiceEndpoint": false
        }
    ],
    "ipRules": [
        {
            "value": "****"
        }
    ]
},
```
any input regarding this issue?

**Comment by apsi15** [comment]
any update on this?

**Comment by stephybun** [comment]
Fixed by #16144
